exercises/giovanni_disiena/ex01_basics/src/main.cpp

```diff
@@ -0,0 +1,160 @@
+#include "main.h"
+
+int main(int argc, char **argv) {
+	// prompt user for filename
+	std::cout << "Please enter the name of a file you wish to read: ";
+	std::string file_name;
+	std::cin >> file_name;
+
+	// create map
+	MapType wordCounts;
+
+	//declare file read boolean and unsigned max length
+	bool read_bool;
+	std::size_t max_len = 0;
+
+	// read file
+	try {
+		read_bool = readFile(file_name, wordCounts, max_len);
+	} catch (const char* msg) {
+		std::cerr << msg << std::endl;
+		read_bool = false;
+	}
+
+	//define lambda for comparison logic
+	Comparator compFunctor =
+		[](const Pair& elem1, const Pair& elem2) { return elem1.second > elem2.second; };
+
+	// declare vector
+	std::vector<Pair> ordered_vector;
+
+	// order words
+	orderWords(wordCounts, ordered_vector, compFunctor);
+
+	// write file
+	try {
+		writeFile(ordered_vector, max_len, read_bool);
+	} catch (const char* msg) {
+		std::cerr << msg << std::endl;
+	}
+}
+
+/** Alters string to lower case
+ *
+ * @param word: string to be altered
+ * @return word: string with all capitals lowered
+ *
+ */
+
+std::string str_tolower(std::string& word) {
+	std::transform(word.begin(), word.end(), word.begin(),
+		[](unsigned char c) { return std::tolower(c); });
+	return word;
+}
+
+/** Takes a string from which to clean specified characters from
+ *	appearing after words
+ *
+ * @param word: string to be cleaned
+ * @param cleaned_words: empty vector to store cleaned strings
+ * @return void
+ *
+ */
+
+void cleanWord(std::string& word, std::vector<std::string>& cleaned_words) {
+	static const std::regex string_regex("(\\B'\b|\b'\\B)|([\\.,:;!?()\"-]+)");
+	auto first = std::sregex_token_iterator(word.begin(), word.end(), string_regex, -1);
+	std::copy(first, std::sregex_token_iterator(), std::back_inserter(cleaned_words));
+}
+
+/** Loads a .txt file, checking for success. Subsequently reads the file
+ *	storing all unique words and their numbers of occurrence
+ *
+ * @param filename: name of the input file to read
+ * @param inputmap: map which will store all unique words and instances
+ * @param char_min: default variable to limit the length of words
+ * @return bool: check for read success
+ *
+ */
+
+bool readFile(const std::string& filename, MapType& inputmap, std::size_t& max_len, const std::size_t char_min) {
+	// attempt to open file
+	std::ifstream inFile;
+	inFile.open(filename + ".txt");
+	if (!inFile) {
+		throw "Unable to open requested file";
```

Comment:
It's unusual to throw a string in this way, usually you'd throw some form of exception e.g. `std::runtime_error`